lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -290,11 +281,14 @@ class Object < ObjectAttribute
     end
 
     class Operation < PolicyElement
-      has_and_belongs_to_many :policy_element_associations, class_name: 'PolicyMachineStorageAdapter::ActiveRecord::PolicyElementAssociation', join_table: 'operations_policy_element_associations'
```

Comment:
How does a recursive through look? I did some poking around on the interwebs and didn't stumble across an example.